src/Carnac.Logic/MessageController.cs

```diff
@@ -26,9 +28,13 @@ public KeysController(ObservableCollection<Message> keys, IMessageProvider messa
 
         public void Start()
         {
-            var messageStream = messageProvider.GetMessageStream(keyProvider.GetKeyStream()).Publish().RefCount();
+            var keyStream = keyProvider.GetKeyStream();
+            var messageStream = messageProvider.GetMessageStream(keyStream).Publish();
```

Comment:
This may be my misunderstanding of how this is designed, but why does the MessageController need the two dependencies of keyProvider and messageProvider here.  Instead, can a dependency be injected that does this messageProvider.GetMessageStream(keyStream) for you?   Or doesn't it really matter? :)